modules/services/sketchybar/default.nix

```diff
@@ -0,0 +1,59 @@
+{ config, lib, pkgs, ... }:
+
+let
+  inherit (lib) literalExpression maintainers mdDoc mkEnableOption mkIf mkPackageOptionMD mkOption optionals types;
+
+  cfg = config.services.sketchybar;
+
+  configFile = pkgs.writeScript "sketchybarrc" cfg.config;
+in
+
+{
+
+  meta.maintainers = [
+    maintainers.azuwis or "azuwis"
+  ];
+
+  options.services.sketchybar = {
+    enable = mkEnableOption (mdDoc "sketchybar");
+
+    package = mkPackageOptionMD pkgs "sketchybar" { };
+
+    extraPackages = mkOption {
+      type = types.listOf types.package;
+      default = [ ];
+      description = mdDoc ''
+        Extra packages to add to PATH.
+      '';
+      example = literalExpression "[ pkgs.jq ]";
+    };
+
+    config = mkOption {
+      type = types.lines;
+      default = "";
+      example = literalExpression ''
+        sketchybar --bar height=24
+        sketchybar --update
+        echo "sketchybar configuration loaded.."
+      '';
+      description = mdDoc ''
+        Content of configuration file. If empty(default), configuration file won't be managed.
+
+        See [documentation](https://felixkratz.github.io/SketchyBar/)
+        and [example](https://github.com/FelixKratz/SketchyBar/blob/master/sketchybarrc).
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ cfg.package ];
+
+    launchd.user.agents.sketchybar = {
+      path = [ cfg.package ] ++ cfg.extraPackages ++ [ config.environment.systemPath ];
+      serviceConfig.ProgramArguments = [ "${cfg.package}/bin/sketchybar" ]
+        ++ optionals (cfg.config != "") [ "--config" "${configFile}" ];
+      serviceConfig.KeepAlive = true;
+      serviceConfig.RunAtLoad = true;
```

Comment:
We'll need to set `EnvironmentVariables`.`PATH` so that we can include `jq` and other binaries being used in the config. 

See:
- https://github.com/bphenriques/dotfiles/blob/master/darwin/modules/services/sketchybar/default.nix#L56C9-L56C29 
- Usage: https://github.com/bphenriques/dotfiles/blob/master/darwin/default.nix#L86

Reply:
Example of a sketchybar config that requires additional binaries in the `PATH`: https://github.com/bphenriques/dotfiles/blob/master/home/config/sketchybar/config/plugins/bluetooth.sh#L5

The plugin won't work without `jq` in the `PATH`

Reply:
The `PATH` environment variable gets set here already: https://github.com/LnL7/nix-darwin/pull/595/files#diff-c768f292b1281301e751d244aedf4a9a1bf53b978f81a4a914989bdc042244c1R52

Reply:
Nice 👍 I very much prefer that approach